srvmon

```diff
@@ -11,6 +11,19 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+# macOS compatibility
+unameOut="$(uname -s)"
+case "${unameOut}" in
+  Darwin*)
+    datecmd="gdate"
+    command -v "${datecmd}">/dev/null 2>&1||{ echo "Error: install ${datecmd} first, e.g., via brew install coreutils" >&2; exit 1; }
```

Comment:
This command check is used three times, I think it's a candidate for a desperate function. Which you are allowed to write out, not the small ternary if form. Which IMHO improves readability. Like so:

```

command_exists() {
    # check if command exists and fail otherwise
    command -v "$1" >/dev/null 2>&1
    if [[ $? -ne 0 ]]; then
        echo "I require $1 but it's not installed. Please install it. Aborting."
        exit 1
    fi
}
```

Then you can do:

    command_exists curl

Or even a small loop with several dependencies.